core/src/test/java/io/jenkins/pluginhealth/scoring/probes/NumberOfOpenIssuesProbeTest.java

```diff
@@ -0,0 +1,235 @@
+                )
+            )
+        ));
+
+        when(plugin.getName()).thenReturn(pluginName);
+        when(plugin.getScm()).thenReturn(scmLink);
+        when(ctx.getRepositoryName(scmLink)).thenReturn(Optional.of(repository));
+        when(restTemplate.getForEntity(anyString(), anyString().getClass())).thenReturn(responseEntity);
+        when(responseEntity.getBody()).thenReturn(jsonString);
+        when(jsonNode.get(anyString())).thenReturn(new ObjectMapper().readTree(jsonString));
+
+        final NumberOfOpenIssuesProbe probe = getSpy();
+
+        assertThat(probe.apply(plugin, ctx))
+            .usingRecursiveComparison()
+            .comparingOnlyFields("id", "status", "message")
+            .isEqualTo(ProbeResult.success(NumberOfOpenIssuesProbe.KEY, "67 open issues found in JIRA."));
+        verify(probe, atMostOnce()).doApply(plugin, ctx);
+    }
+
+    @Test
+   void shouldBeAbleToFindNumberOfOpenIssuesOnlyInGH() throws IOException {
+        final String pluginName = "cloudevents";
+        final String repository = "jenkinsci/" + pluginName + "-plugin";
+        final String scmLink = "https://github.com/" + repository;
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final GitHub gh = mock(GitHub.class);
+        final GHRepository ghRepository = mock(GHRepository.class);
+
+        when(plugin.getName()).thenReturn(pluginName);
+        when(plugin.getDetails()).thenReturn(
+            Map.of(
+                SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+                UpdateCenterPluginPublicationProbe.KEY, ProbeResult.success(UpdateCenterPluginPublicationProbe.KEY, "")
+            )
+        );
+
+        when(plugin.getScm()).thenReturn(scmLink);
+        when(ctx.getUpdateCenter()).thenReturn(new UpdateCenter(
+            Map.of(),
+            Map.of(),
+            List.of(),
+            List.of(
+                Map.of(
+                    "reportUrl",
+                    "https://github.com/" + repository + "/issues/new/choose",
+                    "type",
+                    "github",
+                    "viewUrl",
+                    "https://github.com/" + repository + "/issues"
+                )
+            )
+        ));
+        when(ctx.getGitHub()).thenReturn(gh);
+        when(ctx.getRepositoryName(plugin.getScm())).thenReturn(Optional.of(repository));
+        when(gh.getRepository(repository)).thenReturn(ghRepository);
+        when(ghRepository.getOpenIssueCount()).thenReturn(6);
+
+        final NumberOfOpenIssuesProbe probe = getSpy();
+
+        assertThat(probe.apply(plugin, ctx))
+            .usingRecursiveComparison()
+            .comparingOnlyFields("id", "status", "message")
+            .isEqualTo(ProbeResult.success(NumberOfOpenIssuesProbe.KEY, "6 open issues found in GitHub."));
+        verify(probe, atMostOnce()).doApply(plugin, ctx);
```

Comment:
What does `atMostOnce` method does?

```suggestion
        verify(probe).doApply(plugin, ctx);
```